app/Http/Controllers/Common/CronController.php

```diff
@@ -438,6 +438,70 @@ public function postRenewalNotify()
         }
     }
 
+    /**
+     * Deletes old invoices based on specified criteria.
+     *
+     * This function checks if invoices should be deleted, retrieves old invoices,
+     * and deletes invoices that meet specific conditions.
+     *
+     * @return void
+     */
+    public function invoicesDeletion()
+    {
+        if ($this->shouldDeleteInvoices()) {
+            $days = ExpiryMailDay::value('invoice_days');
+            $dueInvoices = $this->getOldInvoices($days);
+
+            foreach ($dueInvoices as $invoice) {
+                if ($this->canDeleteInvoice($invoice)) {
+                    $this->deleteInvoice($invoice);
+                }
+            }
+        }
+    }
+
+    private function shouldDeleteInvoices()
+    {
+        return StatusSetting::value('invoice_deletion_status') == 1;
+    }
+
+    private function getOldInvoices($days)
+    {
+        $date = Carbon::now()->subDays($days);
+        $oldInvoices = Invoice::where('status', 'pending')
```

Comment:
Could also do this $oldInvoices = Invoice::where('status', 'pending')
                      ->where('date', '<', $date)
                      ->with(['invoiceItem', 'orderRelation'])
                      ->get();